plugins/hsqldb-adapter/src/main/java/org/polypheny/db/hsqldb/stores/HsqldbStore.java

```diff
@@ -212,48 +193,39 @@ protected String getTypeString( PolyType type ) {
         if ( type.getFamily() == PolyTypeFamily.MULTIMEDIA ) {
             return "BLOB(" + RuntimeConfig.UI_UPLOAD_SIZE_MB.getInteger() + "M)";
         }
-        switch ( type ) {
-            case BOOLEAN:
-                return "BOOLEAN";
-            case VARBINARY:
-                return "VARBINARY";
-            case TINYINT:
-                return "TINYINT";
-            case SMALLINT:
-                return "SMALLINT";
-            case INTEGER:
-                return "INT";
-            case BIGINT:
-                return "BIGINT";
-            case REAL:
-                return "REAL";
-            case DOUBLE:
-                return "FLOAT";
-            case DECIMAL:
-                return "DECIMAL";
-            case VARCHAR:
-                return "VARCHAR";
-            case DATE:
-                return "DATE";
-            case TIME:
-                return "TIME";
-            case TIMESTAMP:
-                return "TIMESTAMP";
-            case ARRAY:
-                return "LONGVARCHAR";
-            case JSON:
-            case NODE:
-            case EDGE:
-            case DOCUMENT:
-                return "LONGVARCHAR";
-        }
-        throw new RuntimeException( "Unknown type: " + type.name() );
+        return switch ( type ) {
+            case BOOLEAN -> "BOOLEAN";
+            case VARBINARY -> "VARBINARY";
+            case TINYINT -> "TINYINT";
+            case SMALLINT -> "SMALLINT";
+            case INTEGER -> "INT";
+            case BIGINT -> "BIGINT";
+            case REAL -> "REAL";
+            case DOUBLE -> "FLOAT";
+            case DECIMAL -> "DECIMAL";
+            case VARCHAR -> "VARCHAR";
+            case DATE -> "DATE";
+            case TIME -> "TIME";
+            case TIMESTAMP -> "TIMESTAMP";
+            case ARRAY -> "LONGVARCHAR";
+            case TEXT -> "VARCHAR(200000)"; // clob can sadly not be used as pk which puts arbitrary limit on the value
```

Comment:
There is an alias type LONGVARCHAR in hsqldb (also used for Document and Graph). Per default, this maps to VARCHAR, however, there is also the option to map it to CLOB. 
`SET DATABASE SQL LONGVAR IS LOB { TRUE | FALSE }`. 
At the end, both is a tradeoff because CLOB is slower for small string than VARCHAR. 
HSQLDB max length for varchar is (according to the documentation): 2147483648 (2 * 1024 *1024 *1024)

Reply:
Yes, but Hsqldb cannot handle CLOBs as primary keys and throws an error. This makes it rather inflexible as a substitute. For example Document has to support all values as _id, which we use as substitute PK, similar for nested array substitution, which limits what can be used as pk or we need as substitution for the substitution, which gets ugly. Therefore, while less ideal I prefer VARCHAR here.